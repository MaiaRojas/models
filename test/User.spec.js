const mongoose = require('mongoose');
const { User } = require('../')(mongoose);


describe('User', () => {
  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
    await User.createIndexes();
  });

  it('should ony require uid, email and name??', () => {
    const user = new User({
      uid: 'abcdefghijklmnopqrstuvwxyz',
      email: 'someone@somewhere.com',
      name: 'Someone',
    });
    return user.save()
      .then((result) => {
        expect(result.uid).toBe('abcdefghijklmnopqrstuvwxyz');
        expect(result.email).toBe('someone@somewhere.com');
        expect(result.name).toBe('Someone');
        expect(Array.isArray(result.roles)).toBe(true);
        expect(result.roles.length).toBe(0);
      });
  });
});
