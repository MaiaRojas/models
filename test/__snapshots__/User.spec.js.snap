// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should fail given an AcademicProfile comments with missing cohortProject 1`] = `"User validation failed: academicProfile.comments.0.cohortProject: Cast to ObjectID failed for value \\"cohortProject._id\\" at path \\"cohortProject\\", academicProfile.comments.0.text: Path \`text\` is required., academicProfile: Validation failed: comments.0.cohortProject: Cast to ObjectID failed for value \\"cohortProject._id\\" at path \\"cohortProject\\", comments.0.text: Path \`text\` is required."`;

exports[`User should fail given an AcademicProfile comments with missing text 1`] = `"User validation failed: academicProfile.comments.0.text: Path \`text\` is required., academicProfile: Validation failed: comments.0.text: Path \`text\` is required."`;

exports[`User should fail given an AcademicProfile comments with type other 1`] = `"User validation failed: academicProfile.comments.0.type: \`other\` is not a valid enum value for path \`type\`., academicProfile: Validation failed: comments.0.type: \`other\` is not a valid enum value for path \`type\`."`;
